subcommands/docker/cmd.go

```diff
@@ -130,3 +140,31 @@ func RunCredsHelper() int {
 	os.Stdout.Write(bytes)
 	return 0
 }
+
+// Find an entry in the PATH we can write to. For example, on MacOS git is
+// installed under /usr/bin but even root can't write to that because of
+// filesystem protection logic they have.
+func findWritableDirInPath(gitPath string) string {
```

Comment:
i would have done it this way also, but after a little thinking - I think we should move this to subcommands/common*.go and have both docker/git use the same thing.